src/interface/blas3/backend/amd_gpu.hpp

```diff
@@ -33,8 +33,14 @@ namespace backend {
 template <bool _t_a, bool _t_b, bool s_a, bool s_b, bool is_beta_zero,
           typename sb_handle_t, typename container_0_t, typename container_1_t,
           typename container_2_t, typename element_t, typename index_t>
+#ifndef BLAS_ENABLE_HALF
 typename std::enable_if<is_sycl_scalar<element_t>::value,
                         typename sb_handle_t::event_t>::type
+#else
+typename std::enable_if<is_half<element_t>::value ||
```

Comment:
If you remove the `BLAS_ENABLE_HALF` guard from `is_half` in `include/blas_meta.h`, you can also remove this `#ifndef` pragma keeping just the false case.

Reply:
Yes that's a good idea thank you!